packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRange.ts

```diff
@@ -147,13 +147,31 @@ export async function beaconBlocksMaybeBlobsByRange(
     }, [] as number[]);
 
     const dataColumnRequest = {...request, columns};
-    const [allBlocks, allDataColumnSidecars] = await Promise.all([
-      // TODO-das: investigate why partialDownload blocks is empty here
+    const allBlocksPromise =
       partialDownload && partialDownload.blocks.length > 0
-        ? partialDownload.blocks.map((blockInput) => ({data: blockInput.block}))
-        : network.sendBeaconBlocksByRange(peerId, request),
-      columns.length === 0 ? [] : network.sendDataColumnSidecarsByRange(peerId, dataColumnRequest),
-    ]);
+        ? Promise.resolve(partialDownload.blocks.map((blockInput) => ({data: blockInput.block})))
+        : network.sendBeaconBlocksByRange(peerId, request);
+
+    // DataColumnSidecars may take time for migration from unfinalized to finalized db
+    // So we retry to fetch the data column sidecars once with a delay
+    const allDataColumnSidecarsPromise =
+      columns.length === 0
+        ? Promise.resolve([])
+        : retry(() => network.sendDataColumnSidecarsByRange(peerId, dataColumnRequest), {retries: 2, retryDelay: 1000});
+
+    // We want to make sure neither of the request does not fail because of other so handled it with `Promise.allSettled`
+    const [blockResp, dataColumnResp] = await Promise.allSettled([allBlocksPromise, allDataColumnSidecarsPromise]);
```

Comment:
this is not a guarantee we have all DataColumnSidecars we need, should not be a `Promise.allSettled()` here

Reply:
Curious what drove the switch from `Promise.all` to `Promise.allSettled`?  I tend to prefer `all` but would like to know why you made the switch.